lib/integration/fsc.ts

```diff
@@ -1,15 +1,62 @@
 import { TIME_ZONE } from "config/config";
 import { calculateMondayOffset } from "lib/integration/common";
 import { DateTime } from "luxon";
-import { FscWeekSchedule, FscWeekScheduleResponse } from "types/integration/fsc";
+import {
+    DetailedFscWeekSchedule,
+    FscActivityDetail,
+    FscActivityDetailResponse,
+    FscWeekSchedule,
+    FscWeekScheduleResponse,
+} from "types/integration/fsc";
 
 function fscWeekScheduleUrl(fromDate: DateTime) {
     return `https://fsc.no/api/v1/businessunits/8/groupactivities?period_start=${fromDate.toUTC()}&period_end=${fromDate
         .plus({ week: 1 })
         .toUTC()}`;
 }
 
-export async function fetchFscWeekSchedule(weekOffset: number): Promise<FscWeekSchedule> {
+function fscActivityDetailUrl(activityId: number) {
+    return `https://fsc.no/api/v1/products/groupactivities/${activityId}`;
+}
+
+async function fetchDetailedFscWeekSchedule(fscWeekSchedule: FscWeekSchedule): Promise<DetailedFscWeekSchedule> {
+    const cachedActivityDetails: FscActivityDetail[] = [];
```

Comment:
What about using a map indexed by activity id for quicker access?

Reply:
Sure, fixed in https://github.com/mathiazom/rezervo-web/pull/54/commits/2de2506afc6864fc2fe940f62234f2615bce7d4e. I think the bottleneck is mostly the time it takes to fetch, but yeah this is a nice optimization

Reply:
I looked at this a bit more, and I made some changes so that these details are fetched in parallel. Now, we have reduced the build time from 9000ms to ~3800ms. This is on par with the fetch time for the Sit schedule.

For reference: https://github.com/mathiazom/rezervo-web/pull/54/commits/d6d004ae41239cab99a5e9b35b595e0f306e10db

Reply:
#### Before
<img width="658" alt="Screenshot 2023-09-02 at 20 15 17" src="https://github.com/mathiazom/rezervo-web/assets/26925695/81c12e67-9a61-4f7e-a80d-fc5d1656fa27">

#### After
<img width="658" alt="Screenshot 2023-09-02 at 20 14 57" src="https://github.com/mathiazom/rezervo-web/assets/26925695/5c39c2e8-8247-4d16-a88c-600f0c329779">